include/ddc/print.hpp

```diff
@@ -0,0 +1,358 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <type_traits>
+#include <utility>
+
+#include "ddc/chunk_span.hpp"
+#include "ddc/discrete_domain.hpp"
+
+#if defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG)
+#    include <cxxabi.h>
+#endif
+
+namespace ddc {
+namespace detail {
+struct ChunkPrinter
+{
+    static int const threshold = 10;
+    // If this ever becomes modifiable by the user, we need to ensure that
+    // edgeitems < (threshold / 2) stays true.
+    static int const edgeitems = 3;
+
+    std::stringstream ss;
+    explicit ChunkPrinter(std::ostream const& os)
+    {
+        ss.copyfmt(os);
+    }
+
+    static std::ostream& alignment(std::ostream& os, int level)
+    {
+        for (int i = 0; i <= level; ++i) {
+            os << ' ';
+        }
+        return os;
+    }
+
+    template <class T>
+    std::size_t get_element_width(T& elem)
+    {
+        ss.seekp(0);
+        ss << elem;
+        return ss.tellp();
+    }
+
+    template <class T>
+    void display_aligned_element(std::ostream& os, T& elem, std::size_t largest_element)
+    {
+        std::size_t const elem_width = get_element_width(elem);
+
+        for (int i = 0; i < largest_element - elem_width; ++i) {
+            os << " ";
+        }
+        os << elem;
+    }
+
+    template <class ElementType, class Extents, class Layout, class Accessor>
+    std::ostream& base_case_display(
+            std::ostream& os,
+            Kokkos::mdspan<ElementType, Extents, Layout, Accessor> const& s,
+            std::size_t largest_element,
+            std::size_t beginning,
+            std::size_t end,
+            std::size_t extent)
+    {
+        for (std::size_t i0 = beginning; i0 < end; ++i0) {
+            display_aligned_element(os, s[i0], largest_element);
+            if (i0 < extent - 1) {
+                os << " ";
+            }
+        }
+        return os;
+    }
+
+    template <class ElementType, class Extents, class Layout, class Accessor, std::size_t... Is>
+    std::ostream& recursive_display(
+            std::ostream& os,
+            Kokkos::mdspan<ElementType, Extents, Layout, Accessor> const& s,
+            int level,
+            std::size_t largest_element,
+            std::size_t beginning,
+            std::size_t end)
```

Comment:
Can you pass a `std::integer_sequence` ? This should allow template deduction simplifying its usage.